packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec

```diff
@@ -177,18 +179,25 @@ EOF
 %{foreman_proxy_bundlerd_dir}/%{plugin_name}.rb
 %exclude %{gem_cache}
 %{gem_spec}
-%{foreman_proxy_bundlerd_dir}/smart_proxy_salt_core.rb
-%{smart_proxy_dynflow_bundlerd_dir}/smart_proxy_salt_core.rb
 %config(noreplace) %{salt_config_dir}/foreman.yaml
 %config %{_root_sysconfdir}/cron.d/%{gem_name}
 %exclude %{gem_instdir}/etc
 %exclude %{gem_instdir}/cron
+%{salt_proxy_runners_dir}
+%{salt_proxy_reactors_dir}
+%dir %attr(-,foreman-proxy,foreman-proxy) %{salt_state_grains_dir}
+%ghost %{salt_state_grains_dir}/autosign_key
```

Comment:
By marking it as %ghost it looks like it's not included in the final RPM. 4.0.0 can't deal with that, so this is still something that should be addressed.

Reply:
So, do we include it without ghosting the file in 4.0.0 and make another release 4.0.1? That 4.0.1 release ghosts the file, does not include it in the RPM direclty, and creates it programmatically with the Smart Proxy instead.

Reply:
The alternative is to add a `%post` section:
```spec
%post
if [ ! -f %{salt_state_grains_dir}/autosign_key ] ; then
  touch %{salt_state_grains_dir}/autosign_key
  chmod 644 %{salt_state_grains_dir}/autosign_key
  chown foreman-proxy:foreman-proxy %{salt_state_grains_dir}/autosign_key
fi
```

I'm not sure the chmod is needed, though for security I'd expect it to not be world readable. key implies there are secrets in there.

I also think right now you don't have the correct ownership on the file and it ends up being unwriteable for the `foreman-proxy` user.

Reply:
LGTM, especially to add the permissions with chown and manage access. Previously, the user access was set when the folder has been created.